docs/Test.md

```diff
@@ -0,0 +1,150 @@
+# Test
+
+Tests serve many different purposes. They can:
+
+- Validate the requirements/expectations of stakeholders are met
+- That an end user can achieve their goal
+- Act as documenatation/reference material
+- Confirm integration works as expected between areas of function
+- Validate we do not regress functionality or behaviour we care about
+
+We want to enable and achieve all of these goals in the UI testing done in this
+repo. To this end, we require, depending on the functionality being added or
+changed, different types of tests. The types of test expected, and the
+rationale for these are detailed below.
+
+## Style of test
+
+We have two styles of test - `Behavioural` and `Functional`.
+
+`Behavioural` tests focus on exactly that - behaviour. They do not target
+individual functons, but instead test an end to end behaviour (and all the
+functions which contribute to it), and validate that for a given set of actions
+or inputs, that the expected output/end goal is met. These behaviours should
+focus on the user of the component. Depending on the type of code being
+developed, the user may be another developer (using the component for example)
+or the end user of the UI.
+
+`Functional` tests conversely target individual functions in a more traditional
+unit test style - validating that for a given input, the expected output is
+returned.
+
+Depending on the code being covered, the tools used to enable these test styles
+may be different, as detailed below.
+
+### Functional test style
+
+Code in the following directories are expected to be tested by `functional`
+test methods, using _Jest_ as a test runner, using standard Jest test assets,
+such as `describe`, `it` and `expect`:
+
+- `client/Hooks`
+- `client/Contexts`
+
+### Behavioural test style
```

Comment:
Where are we putting the behavioral and functional tests?  Are they going to live alongside the components, elements, hooks, contexts, and panel code in src?   Or are we going to put them somewhere else?

Reply:
They'll go alongside each component as a regular test would. The split between behavioural/functional is more of a guidance of what approach should be taken for each component. I'll add a piece that says this is where they should go